EmotiBitLedController.h

```diff
@@ -0,0 +1,106 @@
+/**************************************************************************/
+/*!
+    @file     EmotiBitLedController.h
+
+    This class handles LEDs on EmotiBit.
+
+    EmotiBit invests time and resources providing this open source code,
+    please support EmotiBit and open-source hardware by purchasing
+    products from EmotiBit!
+
+    Written by Nitin Nair for EmotiBit.
+
+    BSD license, all text here must be included in any redistribution
+*/
+/**************************************************************************/
+#ifndef EMOTIBIT_LED_CONTROLLER
+#define EMOTIBIT_LED_CONTROLLER
+#include <Arduino.h>
+#include "EmotiBitVersionController.h"
+#include "EmotiBit_NCP5623.h"
+
+class EmotiBitLedController
+{
+public:
+    /*!
+     * @brief enumerator for LEDs on EmotiBit 
+    */
+    enum Led
+    {
+        RED = 0,
+        BLUE, 
+        YELLOW,
+        length 
+    };
+
+    /*!
+     * @brief struct to hold current state of LEDs on EmotiBit
+    */
+    struct LedStatus
+    {
+        bool state = false;
+        bool isChanged = false;
```

Comment:
Here was my thought process:
1. Since we now have the LED controller, it becomes the user facing element and all `seek/update` calls will only go to the controller, so I thought it was necessary for the controller to hold the state of LEDs.
2. Yes, there can be a mismatch in the LED states as stored in the controller and the driver, but, the state update is only unidirectional. The EmotiBit class updates the LedController which in turn updates the driver. Since EmotiBit only has access to the LedController, the multiple source of truth although present is not realized.
3. Furthermore, I think we we want to remove the MSOT, we should drop the storage element in the driver. And use it purely for communicating with the IC. I think, in the future, that is the way to proceed. Hence, I am going to leave it as is now because removing it from the LedController may go against the final change we want to make.

Let me if there is something I missed in the reasoning.

Reply:
After thinking about it, I think I agree it should be pushed down to the driver 😅 . Below is the reason.
TLDR;
If we keep the state storage in the driver, the driver can be used asynchronously, but if the storage lives in a wrapper class above the driver, then the driver can only be used synchronously without additionally creating a wrapper class.

![image](https://github.com/user-attachments/assets/db657ab1-f247-4ec9-ba73-3086786db60b)

Reply:
Moved to issue: https://github.com/EmotiBit/EmotiBit_FeatherWing/issues/317